File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/events/SLInventoryNewContentsEvent.java
package com.lumiyaviewer.lumiya.slproto.events;

import java.util.UUID;

public class SLInventoryNewContentsEvent {
    public int newFolderCount = 0;
    public int newItemCount = 0;
    public UUID firstParentUUID = null;
    public String firstItemName = null;
    public boolean firstIsFolder = false;

    public void AddItem(boolean firstIsFolder, UUID uuid, String firstItemName) {
        if (firstIsFolder) {
            this.newFolderCount++;
        } else {
            this.newItemCount++;
        }
        if (this.firstParentUUID == null) {
            this.firstIsFolder = firstIsFolder;
            this.firstParentUUID = uuid;
            this.firstItemName = firstItemName;
        } else if (firstIsFolder && (!this.firstIsFolder)) {
            this.firstIsFolder = firstIsFolder;
            this.firstParentUUID = uuid;
            this.firstItemName = firstItemName;
        }
    }

    public boolean isEmpty() {
        return this.newFolderCount == 0 && this.newItemCount == 0;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/slproto/events/SLInventoryNewContentsEvent.kt
package com.lumiyaviewer.lumiya.slproto.events

import java.util.UUID

class SLInventoryNewContentsEvent {
    @JvmField
    var newFolderCount: Int = 0

    @JvmField
    var newItemCount: Int = 0

    @JvmField
    var firstParentUUID: UUID? = null

    @JvmField
    var firstItemName: String? = null

    @JvmField
    var firstIsFolder: Boolean = false

    fun AddItem(firstIsFolder: Boolean, uuid: UUID, firstItemName: String) {
        if (firstIsFolder) {
            newFolderCount++
        } else {
            newItemCount++
        }
        if (firstParentUUID == null) {
            this.firstIsFolder = firstIsFolder
            firstParentUUID = uuid
            this.firstItemName = firstItemName
        } else if (firstIsFolder && !this.firstIsFolder) {
            this.firstIsFolder = firstIsFolder
            firstParentUUID = uuid
            this.firstItemName = firstItemName
        }
    }

    fun isEmpty(): Boolean {
        return newFolderCount == 0 && newItemCount == 0
    }
}
